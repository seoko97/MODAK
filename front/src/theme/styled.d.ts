import "styled-components";
import { bp } from ".";

declare module "styled-components" {
  export interface DefaultTheme {
    BP: typeof bp;
    BAKCGROUND_COLOR: {
      PRIMARY_COLOR: string;
      SECONDARY_COLOR: string;
      THIRDARY_COLOR: string;
      RGBA: string;
    };
    FONT_COLOR: {
      PRIMARY_COLOR: string;
      SECONDARY_COLOR: string;
    };
    BORDER_COLOR: {
      PRIMARY_COLOR: string;
    };
  }
}
