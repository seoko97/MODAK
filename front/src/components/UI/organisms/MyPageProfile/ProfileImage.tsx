import { checkUrl } from "@src/lib/checkUrl";
import React from "react";
import Style from "./styles";

interface Props {
  profileImg: string;
}
const ProfileImage = ({ profileImg }: Props) => (
  <Style.ProfileImage>
    <Style.Img src={checkUrl(profileImg)} />
  </Style.ProfileImage>
);
export default ProfileImage;
