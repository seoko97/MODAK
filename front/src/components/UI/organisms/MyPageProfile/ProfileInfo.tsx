import HeartIcon from "@src/components/icons/HeartIcon";
import PencilIcon from "@src/components/icons/PencilIcon";
import UserUpdate from "@src/components/modals/UserUpdateForm";
import React from "react";
import Title from "../../atoms/Title";
import Style from "./styles";

interface Props {
  nickname: string;
  onOpen: () => void;
  isOpen: boolean;
  onClose: () => void;
  intro: string;
  displayEditButton: boolean;
}

const IconComponent = () => (
  <Style.IconBox>
    <Style.Icons>
      <PencilIcon size={13} /> 10
    </Style.Icons>
    <Style.Icons>
      <HeartIcon size={13} /> 10
    </Style.Icons>
  </Style.IconBox>
);

const ProfileInfo = ({ nickname, onOpen, isOpen, onClose, intro, displayEditButton }: Props) => (
  <Style.ProfileInfo>
    <Style.UserName>
      <Title size={14}>{nickname}</Title>
      {displayEditButton && (
        <Style.EditProfile onClick={onOpen}>
          <PencilIcon size={13} />
        </Style.EditProfile>
      )}
      {isOpen && <UserUpdate onClick={onClose} />}
    </Style.UserName>
    <p>{intro}</p>
    <IconComponent />
  </Style.ProfileInfo>
);

export default ProfileInfo;
