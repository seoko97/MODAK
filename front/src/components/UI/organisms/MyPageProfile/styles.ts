import styled, { css } from "styled-components";

// --- 프로필 컨테이너 ---
const Profile = styled.figure`
  display: flex;
  gap: 30px;
  position: relative;
  margin-bottom: 10px;
  @media (max-width: ${({ theme }) => theme.BP.MOBILE}) {
    flex-direction: column;
    align-items: center;
  }
`;

// --- 프로필 이미지 ---
const ImageBoxStyle = css`
  width: 130px;
  height: 130px;
  @media (max-width: ${({ theme }) => theme.BP.TABLET}) {
    width: 120px;
    height: 120px;
  }
`;

const ProfileImage = styled.div`
  ${ImageBoxStyle}
`;

const Img = styled.img`
  ${ImageBoxStyle}
  border-radius: 50%;
`;

// --- 프로필 정보 ---
const ProfileInfo = styled.figcaption`
  color: ${({ theme }) => theme.FONT_COLOR.PRIMARY_COLOR};
  & h2,
  p {
    margin-bottom: 10px;
  }
`;

const UserName = styled.div`
  display: flex;
  align-items: baseline;
  gap: 6px;
`;

const IconBox = styled.div`
  display: flex;
`;

const Icons = styled.div`
  display: flex;
  box-sizing: border-box;
  padding: 5px;
  gap: 10px;
  justify-content: center;
  align-items: baseline;
  font-size: 14px;
`;

const EditProfile = styled.button`
  cursor: pointer;
  border: none;
  background-color: transparent;
`;

export default {
  EditProfile,
  IconBox,
  Icons,
  ImageBoxStyle,
  Img,
  Profile,
  ProfileImage,
  ProfileInfo,
  UserName,
};
