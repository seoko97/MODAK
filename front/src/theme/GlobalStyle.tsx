import { createGlobalStyle } from "styled-components";
import { reset } from "styled-reset";

const GlobalStyle = createGlobalStyle`
    ${reset}
      body {
        overflow-y: overlay;
        margin: 0;

        font-size: 14px;
        line-height: 1.5715;
      }

      body,
      #__next {
        width: 100%;
        transition: 0.3s background-color;
        background-color: ${({ theme }) => theme.BAKCGROUND_COLOR.PRIMARY_COLOR};
      }

      h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        margin-top: 0;
      }
      a {
        color: inherit;
        text-decoration: none;
        outline: none;
      }

      * {
        font-family: "Quicksand", "Noto Sans KR", sans-serif !important;
        box-sizing: border-box;
      }
`;

export default GlobalStyle;
