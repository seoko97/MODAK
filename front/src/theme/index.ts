import { DefaultTheme } from "styled-components";

export const bp = {
  HDPC: "1200px",
  PC: "980px",
  TABLET: "768px",
  MOBILE: "480px",
};

export const lightTheme: DefaultTheme = {
  BP: bp,
  BAKCGROUND_COLOR: {
    PRIMARY_COLOR: "#ffffff",
    SECONDARY_COLOR: "#23374D",
    THIRDARY_COLOR: "#23374D",
    RGBA: "rgba(255, 255, 255, 0.4)",
  },
  FONT_COLOR: {
    PRIMARY_COLOR: "#212529",
    SECONDARY_COLOR: "#1b2e46",
  },
  BORDER_COLOR: {
    PRIMARY_COLOR: "#25282B",
  },
};

export const darkTheme: DefaultTheme = {
  BP: bp,
  BAKCGROUND_COLOR: {
    PRIMARY_COLOR: "#25282B",
    SECONDARY_COLOR: "#444444",
    THIRDARY_COLOR: "#444444",
    RGBA: "rgba(37, 40, 43, 0.4)",
  },
  FONT_COLOR: {
    PRIMARY_COLOR: "#f8f9fa",
    SECONDARY_COLOR: "#151516",
  },
  BORDER_COLOR: {
    PRIMARY_COLOR: "#ffffff",
  },
};
