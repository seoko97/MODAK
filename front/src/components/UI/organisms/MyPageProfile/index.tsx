import React from "react";
import useModal from "@src/hooks/useModal";
import { useAppSelector } from "@src/store/configureStore";
import { EditUserData } from "@src/types/apis/user";
import ProfileImage from "./ProfileImage";
import ProfileInfo from "./ProfileInfo";
import Style from "./styles";

const MyPageProfile = (): React.ReactElement => {
  const { me, userInfo } = useAppSelector((state) => state.user);
  const [isOpen, onOpen, onClose] = useModal();
  const { nickname, profileImg, intro } = me as EditUserData;
  const displayEdit = me?._id === userInfo?._id;

  return (
    <Style.Profile>
      <ProfileImage profileImg={profileImg} />
      <ProfileInfo
        displayEditButton={displayEdit}
        nickname={nickname}
        onOpen={onOpen}
        isOpen={isOpen}
        onClose={onClose}
        intro={intro}
      />
    </Style.Profile>
  );
};

export default MyPageProfile;
