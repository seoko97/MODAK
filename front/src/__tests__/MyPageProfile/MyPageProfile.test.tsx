import React from "react";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "@src/theme/index";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import MyPageProfile from "@src/components/UI/molecules/MypageProfile";

describe("MyPageProfile 테스트", () => {
  it("렌더 확인", () => {
    const view = render(
      <ThemeProvider theme={lightTheme}>
        <MyPageProfile />
      </ThemeProvider>,
    );

    expect(view.container).toMatchSnapshot();
  });

  it("프로필 편집 클릭", () => {
    render(
      <ThemeProvider theme={lightTheme}>
        <MyPageProfile />
      </ThemeProvider>,
    );

    const buttons = screen.getByTitle("EditProfile");
    // userEvent.click(buttons);
  });
});
