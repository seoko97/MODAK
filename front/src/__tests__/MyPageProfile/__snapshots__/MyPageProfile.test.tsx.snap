// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPageProfile 테스트 렌더 확인 1`] = `
<div>
  <figure
    class="sc-dJjYzT jSlgeC"
  >
    <div
      class="sc-hGPBjI ZDlqG"
    >
      <img
        class="sc-dlVxhl iocqpi"
        src="https://images.unsplash.com/photo-1560250097-0b93528c311a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80"
      />
    </div>
    <figcaption
      class="sc-kfPuZi flkfTX"
    >
      <div
        class="sc-fKVqWL gsZNVz"
      >
        <h2
          class="sc-ieecCq jJBbDT"
          size="14"
        >
          현
        </h2>
        <button
          class="sc-cxpSdN fSVwmX"
          title="EditProfile"
        >
          <svg
            class="sc-gsDKAQ dgFSI"
            height="13"
            viewBox="0 0 24 24"
            width="13"
          >
            <path
              d="M14.078 7.061l2.861 2.862-10.799 10.798-3.584.723.724-3.585 10.798-10.798zm0-2.829l-12.64 12.64-1.438 7.128 7.127-1.438 12.642-12.64-5.691-5.69zm7.105 4.277l2.817-2.82-5.691-5.689-2.816 2.817 5.69 5.692z"
            />
          </svg>
        </button>
      </div>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Illo eveniet magnam dolor quasi laboriosam, qui facere deleniti, assumenda eius, expedita quisquam sit. Vitae veniam corrupti voluptatum laboriosam culpa, accusantium quia!
      </p>
      <div
        class="sc-bBHxTw gRBTjE"
      >
        <div
          class="sc-iwjdpV AcPsQ"
        >
          <svg
            class="sc-gsDKAQ dgFSI"
            height="13"
            viewBox="0 0 24 24"
            width="13"
          >
            <path
              d="M14.078 7.061l2.861 2.862-10.799 10.798-3.584.723.724-3.585 10.798-10.798zm0-2.829l-12.64 12.64-1.438 7.128 7.127-1.438 12.642-12.64-5.691-5.69zm7.105 4.277l2.817-2.82-5.691-5.689-2.816 2.817 5.69 5.692z"
            />
          </svg>
           10
        </div>
        <div
          class="sc-iwjdpV AcPsQ"
        >
          <svg
            class="sc-bdvvtL hhoKvZ"
            height="13"
            viewBox="0 0 24 24"
            width="13"
          >
            <path
              d="M6.28 3c3.236.001 4.973 3.491 5.72 5.031.75-1.547 2.469-5.021 5.726-5.021 2.058 0 4.274 1.309 4.274 4.182 0 3.442-4.744 7.851-10 13-5.258-5.151-10-9.559-10-13 0-2.676 1.965-4.193 4.28-4.192zm.001-2c-3.183 0-6.281 2.187-6.281 6.192 0 4.661 5.57 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-4.011-3.097-6.182-6.274-6.182-2.204 0-4.446 1.042-5.726 3.238-1.285-2.206-3.522-3.248-5.719-3.248z"
            />
          </svg>
           10
        </div>
      </div>
    </figcaption>
  </figure>
</div>
`;
